packages/contracts-bedrock/src/L1/OptimismPortal2.sol

```diff
@@ -435,8 +448,8 @@ contract OptimismPortal2 is Initializable, ResourceMetering, ISemver {
     /// @notice Checks if a withdrawal can be finalized. This function will revert if the withdrawal cannot be
     ///         finalized, and otherwise has no side-effects.
     /// @param _withdrawalHash Hash of the withdrawal to check.
-    function checkWithdrawal(bytes32 _withdrawalHash) public view {
-        ProvenWithdrawal memory provenWithdrawal = provenWithdrawals[_withdrawalHash];
+    function checkWithdrawal(bytes32 _withdrawalHash, address _proofSubmitter) public view {
+        ProvenWithdrawal memory provenWithdrawal = provenWithdrawals[_withdrawalHash][_proofSubmitter];
```

Comment:
It may be worth updating the error message for `withdrawal has not been proven yet` to provide some context about the change. Something like `OptimismPortal: withdrawal has not been proven by sender address yet, verify correct proof submitter`. You get the idea.